platform/ios/PolyPodApp/PodApi/Endpoint/Endpoint.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import MessagePack
+
+protocol EndpointProtocol {
+    func send(endpointId: String, payload: String, contentType: String?, authorization: String?, completionHandler: @escaping (String?) -> Void) -> Void
+    func get(endpointId: String, contentType: String?, authorization: String?, completionHandler: @escaping (String?, String?) -> Void) -> Void
+}
+
+protocol EndpointDelegate {
+    func doHandleApproveEndpointFetch(endpointId: String, completion: @escaping (Bool) -> Void) -> Void
+}
+
+protocol EndpointInfoProtocol: Decodable {
```

Comment:
He's just staying consistent with how the other API interfaces are named, I'd prefer to stay consistent for now, rather refactoring it everywhere.

Reply:
I think this is just regarding the Serialization type EndpointInfo, where I tried out removing the Protocol and it works so I'll keep it

Reply:
> He's just staying consistent with how the other API interfaces are named, I'd prefer to stay consistent for now, rather refactoring it everywhere.

I am not sure I understand what you are referring to. This `EndpointInfoProtocol` is describing a value type, for which there is `Struct` type to be used. I have searched through PolyPod and I don't see any other instance for similar use case.

If we want to keep it, then at least `endpointInfoFromId` should return `EndpointInfoProtocol ` instead of `EndpointInfo`. Otherwise it is not useful.